api/api/models/couchbase.js

```diff
@@ -106,320 +114,282 @@ function verifyWorkspace (workspace) {
     if (err) {
-      callback(err, null)
-    } else if (results && results[key]) {
-      callback(null, results[key])
-    } else {
-      callback(null, null)
+      return callback(err)
     }
+    if (results && results[key]) {
+      return callback(null, results[key])
+    }
+    callback()
   })
 }
 
 function showLogs (event, callback) {
   const key = `logs:${event}`
   logs.get(key, (err, results, cas, misses) => {
     if (err) {
-      callback(err, null)
-    } else if (results && results[key]) {
-      callback(null, results[key])
-    } else {
-      callback(null, null)
+      return callback(err)
+    }
+    if (results && results[key]) {
+      return callback(null, results[key])
     }
+    callback()
   })
 }
 
 function showWorkspace (workspace, callback) {
-  const key = `ws:${workspace['project']}:${workspace['workspace']}`
+  const key = `ws:${workspace.project}:${workspace.workspace}`
   const eventDate = Date.now()
   bucket.get(key, (err, results, cas, misses) => {
     if (err) {
-      callback(err, null)
-    } else if (results && results[key]) {
-      callback(null, results)
-    } else {
-      let payload = {}
-      payload[key] = {
+      return callback(err)
+    }
+    if (results && results[key]) {
+      return callback(null, results)
+    }
+    const payload = {
+      [key]: {
         type: 'workspace',
-        project: workspace['project'],
-        workspace: workspace['workspace'],
+        project: workspace.project,
+        workspace: workspace.workspace,
         ref: 'branch:master',
         state: 'new',
         creation: eventDate,
         lastEvents: []
       }
-
-      bucket.insert(payload, (err, cas, existing) => {
-        if (err) {
-          callback(err, null)
-        } else {
-          callback(null, payload)
-        }
-      })
     }
+
+    bucket.insert(payload, (err, cas, existing) => {
+      if (err) {
+        callback(err, null)
+      } else {
+        callback(null, payload)
+      }
+    })
   })
 }
 
 function feedWorkspace (workspace, result, callback) {
-  const key = `ws:${workspace['project']}:${workspace['workspace']}`
-  bucket.get(key, function (err, data) {
+  const key = `ws:${workspace.project}:${workspace.workspace}`
+  bucket.get(key, (err, data) => {
     if (err) {
-      callback(err, null)
-    } else if (data && data[key]) {
-      logger.info(`${workspace['project']}:${workspace['workspace']}[${data[key]['lastEvents'][0]}] returns (${(data[key].request ? data[key].request.action : 'undefined')}->${result.status})`)
-      let payload = data
-      let request = { }
-      if (payload[key].request) {
-        request = {
-          action: payload[key].request.action,
-          ref: (payload[key].request.ref ? payload[key].request.ref : (payload[key].ref ? payload[key].ref : 'unknown')),
-          event: payload[key].request.event
+      return callback(err, null)
+    }
+    if (!data && !data[key]) {
```

Comment:
This should be `(!data || !data[key])` in that case, no?

Reply:
fixing that